concordium-std-derive/src/lib.rs

```diff
@@ -1531,7 +1531,7 @@ fn reject_derive_worker(input: TokenStream) -> syn::Result<TokenStream> {
         }
     };
 
-    let variant_error_conversions = generate_variant_error_conversions(&enum_data, &enum_ident)?;
+    let variant_error_conversions = generate_variant_error_conversions(enum_data, enum_ident)?;
```

Comment:
I am generally not in favor of making unrelated changes in a PR.

However since this is so small it is OK now.

Reply:
It was required for the point of the checklist "The code compiles without warnings". I was just trying to follow the contributing guide, and calling `clippy --all` gave a couple of warnings. So I decided that might as well fix them in separate commit.

Reply:
Right, but I think you are probably using a newer version of rust.

The CI uses 1.53, but of course newer versions expand the set of clippy issues, sometimes a bit too eagerly :)